janus_server/src/datastore.rs

```diff
@@ -2973,14 +2983,34 @@ pub mod models {
         pub(crate) state: CollectJobState<L, A>,
     }
 
+    impl<const L: usize, A: vdaf::Aggregator<L>> CollectJob<L, A>
+    where
+        for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
+    {
+        /// Create a new `CollectJob` in the `Start` state, with a random collect job ID.
+        pub(crate) fn new(
+            task_id: TaskId,
+            batch_interval: Interval,
+            aggregation_param: &A::AggregationParam,
```

Comment:
nit: this should take `A::AggregationParam` rather than `&A::AggregationParam`; the idea is that this function needs to own an `A::AggregationParam`. It currently does so by `clone`ing a caller-provided reference. But if the caller happens to already own an aggregation param they want to pass in, writing the constructor forces a `clone` where otherwise one would not be necessary. And if the caller only has a reference, they can call `clone` themselves.